### ISSUE TITLE
Spike errors out with empty data using bokeh.

### ISSUE THREAD

**Issue by majidaldo** [ioam/holoviews#1497]
```
hv.notebook_extension('bokeh')
hv.Curve([1,2,3])<<hv.Spikes([])
```
```
C:\Users\Majid\Miniconda\envs\schedule34\lib\site-packages\holoviews\plotting\bokeh\chart.py in get_data(self, element, ranges, empty)
    546             height = self.spike_length
    547             xs, ys = zip(*((np.array([x[0], x[0]]), np.array([pos+height, pos]))
--> 548                            for x in element.array(dims[:1])))
    549 
    550         if not empty and self.invert_axes: xs, ys = ys, xs

ValueError: not enough values to unpack (expected 2, got 0)
```
the mpl display doesn't error but it creates a display when there is no data. might be an issue with empty data handling in hv in general. i don't think the user should worry about empty data.

**Comment by jlstevens** [comment]
> I don't think the user should worry about empty data.

Agreed. I think most of our elements now allow empty data but clearly there are still places where it isn't working properly. Thanks for reporting this!

**Comment by philippjfr** [comment]
> I think most of our elements now allow empty data but clearly there are still places where it isn't working properly.

Right it's supported by all the Element types but certain plot implementations still throw errors, we should add unit tests instantiating plots with empty elements to see which ones still complain.

**Comment by philippjfr** [comment]
This was fixed in the #1566, closing.
